test/test_torrent.cpp

```diff
@@ -240,18 +242,15 @@ TORRENT_TEST(total_wanted)
 
 	std::vector<char> tmp;
 	bencode(std::back_inserter(tmp), t.generate());
-	auto info = std::make_shared<torrent_info>(tmp, from_span);
+	add_torrent_params p = load_torrent_buffer(tmp);
```

Comment:
One more question for the future. Does it make sense to continue using `std::shared_ptr<torrent_info>` in `add_torrent_params`? Why not switch to `std::optional<torrent_info>`?

Reply:
Is `master` branch the one that can accept such breaking changes I mentioned here? Maybe it makes sense to open the PR, so as not to forget all this?

Reply:
I think it makes sense to use `shared_ptr` because it allows adding it to the session without copying it. Once `torrent_info` is fully immutable, the same copy of it can be used internally as is returned by `torrent_file()`

Reply:
> Once `torrent_info` is fully immutable, the same copy of it can be used internally as is returned by `torrent_file()`

Well, it will make sense.